src/Control/Carrier/Interpret.hs

```diff
@@ -37,7 +38,7 @@ import           Unsafe.Coerce (unsafeCoerce)
 
 -- | A @Handler@ is a function that interprets effects described by @sig@ into the carrier monad @m@.
 newtype Handler sig m = Handler
-  { runHandler :: forall s x . sig (InterpretC s sig m) x -> InterpretC s sig m x }
+  { runHandler :: forall s n x . Monad n => (forall y . n y -> InterpretC s sig m y) -> sig n x -> InterpretC s sig m x }
```

Comment:
This has to take a homomorphism just like `alg`, or else we have to keep `HFunctor` around. I think this is reasonable: the whole idea of `runInterpret` is that it allows you to write “a real algebra” inline.